### PR TITLE
Fix usage of `INPUT_SERVER` variables; fixes #342

### LINKED ISSUE #342: the plugin not working correctly after upgrade with new httpd configuration for 10.0.7
### Describe the bug

After upgrading our instance to glpi 10.0.7 /plugin 1.2.0 including adjusting the httpd config as described in thee gpli 10.0.7 installation guide. I noticed the tasks management and monitoring pages were not displaying correctly. When I reversed the httpd config the pages display properly.

Here are a couple screenshots with the change. The Targets, Agents, Clear list, and Delete selected items links are not clickable and the task status does not display within the task management or monitoring pages.
Screenshot 2023-04-10 at 2 09 18 PM
Screenshot 2023-04-10 at 2 10 53 PM

When I reverse the config the pages behave as before but I get a banner that states "Web server root directory configuration is not safe as it permits access to non-public files. See installation documentation for more details."
Screenshot 2023-04-10 at 2 06 56 PM
Screenshot 2023-04-10 at 2 13 11 PM

### To reproduce

follow https://glpi-install.readthedocs.io/en/latest/prerequisites.html#apache-configuration for the 10.0.7 apache vhost configuration changes. (We do use https rather than http as in the example but that should not matter.)

### Expected behavior

Following the upgrade should not break the plugin

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.7

### GLPIInventory plugin

1.2.0

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

**Comment by trasher** [comment]
There is one remaining use of `SCRIPT_NAME`but for a CLI script. In that case, redirection should not cause any effect, so I guess that's still OK.

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
I will add commit to fix other bugs related to usage of `filter_input(INPUT_SERVER, xxx)`.

**Review by cedric-anne — APPROVED** [review]

File: report/not_queried_recently.php [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 include("../../../inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('FusionInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/not_queried_recently.php:42]:
```suggestion
Html::header(__('GLPIINventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: report/ports_date_connections.php [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 include("../../../inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('FusionInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/ports_date_connections.php:40]:
```suggestion
Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: report/switch_ports.history.php [hunk 3]
```diff
@@ -38,13 +38,13 @@
 define('GLPI_ROOT', '../../..');
 include(GLPI_ROOT . "/inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('FusionInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/switch_ports.history.php:41]:
```suggestion
Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: ajax/dropdown_taskjob.php [hunk 4]
```diff
@@ -31,7 +31,7 @@
  * ---------------------------------------------------------------------
  */
 
-if (strpos(filter_input(INPUT_SERVER, "PHP_SELF"), "dropdown_taskjob.php")) {
+if (strpos(filter_input(INPUT_SERVER, "REQUEST_URI"), "dropdown_taskjob.php")) {
```
Comment (trasher) [inline ajax/dropdown_taskjob.php:34]: For URLs specific to the plugin, should not we rather use `plugin_glpiinventory_script_endswith` function?

Comment (trasher) [reply ajax/dropdown_taskjob.php:34]: Anyways, this is not blocking for current PR

File: hook.php [hunk 5]
```diff
@@ -381,9 +381,9 @@ function plugin_glpiinventory_install()
 {
     ini_set("max_execution_time", "0");
 
-    if (basename(filter_input(INPUT_SERVER, "SCRIPT_NAME")) != "cli_install.php") {
+    if (basename(filter_input(INPUT_SERVER, "REQUEST_URI") ?? '') != "cli_install.php") {
```
Comment (trasher) [inline hook.php:384]: In cli mode, `REQUEST_URI` will probably not be set; and HTTP redirection won't cause effects.

File: report/computer_last_inventory.php [hunk 6]
```diff
@@ -37,7 +37,7 @@
 
 include("../../../inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
+Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/computer_last_inventory.php:40]:
```suggestion
Html::header(__('GLPI Inventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

(string with a space already exists in translations)

File: report/not_queried_recently.php [hunk 7]
```diff
@@ -39,7 +39,7 @@
 
 include("../../../inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/not_queried_recently.php:42]:
```suggestion
Html::header(__('GLPI Inventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: report/ports_date_connections.php [hunk 8]
```diff
@@ -37,7 +37,7 @@
 
 include("../../../inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/ports_date_connections.php:40]:
```suggestion
Html::header(__('GLPI Inventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: report/switch_ports.history.php [hunk 9]
```diff
@@ -38,13 +38,13 @@
 define('GLPI_ROOT', '../../..');
 include(GLPI_ROOT . "/inc/includes.php");
 
-Html::header(__('FusionInventory', 'glpiinventory'), filter_input(INPUT_SERVER, "PHP_SELF"), "utils", "report");
+Html::header(__('GLPIInventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```
Comment (trasher) [inline report/switch_ports.history.php:41]:
```suggestion
Html::header(__('GLPI Inventory', 'glpiinventory'), $_SERVER['PHP_SELF'], "utils", "report");
```

File: hook.php (hunk 5 above)
Comment (trasher) [reply hook.php:384]:
```suggestion
    if (basename($_SERVER['SCRIPT_NAME']) != "cli_install.php") {
```

File: ajax/dropdown_taskjob.php (hunk 4 above)
Comment (cedric-anne) [reply ajax/dropdown_taskjob.php:34]: `plugin_glpiinventory_script_endswith` is searching in `front` dir, so it would require to change how it works.

**Pull request merged**